src/components/body/components/chatRoom/ChatRoom.tsx

```diff
@@ -0,0 +1,48 @@
+//style
+import styles from "./ChatRoom.module.css";
+
+//components
+import { ChatArea } from "./components/charArea";
+import { TeamMateProfile } from "./components/teamMateProfile";
+import { AddMessage } from "./components/addMessage/AddMessage";
+import { Spinner } from "../../../spinner/Spinner";
+import { Error } from "../../../error/Error";
+
+//hooks and libs
+import { useChatRoom } from "../../../../hooks/useChatRoom";
+import { useUserId } from "../../../../hooks/UserContext";
+
+//type
+import { User } from "../../../../types/types";
+
+type ChatRoomProps = {
+  activeTeamMate: User;
+};
+
+export const ChatRoom = ({ activeTeamMate }: ChatRoomProps) => {
+  const userId = useUserId();
+
+  const { chatRoom, onAction, loading, error } = useChatRoom(
+    userId,
+    activeTeamMate?.id
+  );
+  return (
+    <div className={styles.chatRoom}>
+      {loading ? (
```

Comment:
Only the chat area is dependent on this chatRoom query.
But currently, every type of query goes into a loading state, whole right pane gets unmounted.

Also, we can disable the add message button until the chats are loaded or if we get an error.